lib/rufo/formatter.rb

```diff
@@ -3075,9 +3075,15 @@ def consume_end_of_line(at_prefix: false, want_semicolon: false, want_multiline:
         # print it, but only if it's not followed by a newline
         if !found_newline && want_semicolon && last != :semicolon
           skip_space
-          case current_token_kind
+          kind = current_token_kind
+          case kind
           when :on_ignored_nl, :on_eof
           else
+            if (kind == :on_kw)
```

Comment:
@mjago How about:

```ruby
else
  if kind == :on_kw
    return if %w[class module def].include?(current_token_value)
  end
  ...
```

or as diff

```diff
diff --git a/lib/rufo/formatter.rb b/lib/rufo/formatter.rb
index a955e2c..38ac60b 100644
--- a/lib/rufo/formatter.rb
+++ b/lib/rufo/formatter.rb
@@ -3079,10 +3079,8 @@ class Rufo::Formatter
           case kind
           when :on_ignored_nl, :on_eof
           else
-            if (kind == :on_kw)
-              %w[class module def].each do |kw|
-                return if current_token_value == kw
-              end
+            if kind == :on_kw
+              return if %w[class module def].include?(current_token_value)
             end
             write "; "
             last = :semicolon
```

Another approach:

```ruby
          case kind
          when :on_ignored_nl, :on_eof
          when :on_kw
            return if %w[class module def].include?(current_token_value)
          else
            write "; "
            last = :semicolon
          end
```

The tests pass although I don't know if they should pass? :confused: